libs/core/src/lib/avatar-group/components/avatar-group-host.component.ts

```diff
@@ -0,0 +1,151 @@
+    /** @hidden */
+    @Input()
+    type: AvatarGroupHostConfig['type'];
+
+    /** @hidden */
+    @Input()
+    orientation: AvatarGroupHostConfig['orientation'];
+
+    /** @hidden */
+    @Input()
+    size: AvatarGroupHostConfig['size'];
+
+    /** @hidden */
+    @Input()
+    resizeEmitter: Observable<ResizeObserverEntry[]> = inject(ResizeObserverDirective).resized;
+
+    /** @hidden */
+    @Input()
+    items: QueryList<AvatarGroupItemDirective>;
+
+    /** @hidden */
+    @ContentChildren(AvatarGroupItemRendererDirective, { descendants: true })
+    portals: QueryList<AvatarGroupItemRendererDirective>;
+
+    /** @hidden */
+    elementRef = inject(ElementRef);
+
+    /** @hidden */
+    hiddenItems = signal<AvatarGroupItemRendererDirective[]>([]);
+
+    /** @hidden */
+    private _destroyRef = inject(DestroyRef);
+
+    /** @hidden */
+    private _cdr = inject(ChangeDetectorRef);
+
+    /** @hidden */
+    private _onChanges$ = new Subject<SimpleChanges>();
+
+    /** @hidden */
+    ngOnChanges(changes: SimpleChanges): void {
+        this._onChanges$.next(changes);
+    }
+
+    /** @hidden */
+    ngAfterViewInit(): void {
+        combineLatest([
+            this.resizeEmitter.pipe(map((entries) => entries[0].contentRect.width)),
+            this.portals.changes.pipe(
+                startWith(this.portals),
+                map((r) => r.toArray())
+            ),
+            this._onChanges$.pipe(startWith({}))
+        ])
+            .pipe(
+                map(([containerWidth, items]) => this.calculateVisibility(containerWidth, items)),
+                delayWhen(() => animationFrames()),
+                takeUntilDestroyed(this._destroyRef)
+            )
+            .subscribe(({ hiddenItems, visibleItems }) => {
+                visibleItems.forEach((item) => item.show());
+                hiddenItems.forEach((item) => item.hide());
+                this._cdr.detectChanges();
+                this.hiddenItems.set(hiddenItems);
+            });
+    }
+
+    /** @hidden */
+    calculateVisibility(
+        containerWidth: number,
+        items: AvatarGroupItemRendererDirective[]
+    ): {
+        hiddenItems: AvatarGroupItemRendererDirective[];
+        visibleItems: AvatarGroupItemRendererDirective[];
+    } {
+        if (this.orientation === 'vertical') {
+            return {
+                visibleItems: items,
+                hiddenItems: []
+            };
+        }
+        const visibleItems = items.filter((i) => i.forceVisibility);
+        const hiddenItems: AvatarGroupItemRendererDirective[] = [];
+        let accWidth = items.reduce((acc, item) => (item.forceVisibility ? acc + item.width : acc), 0);
+        for (const item of items) {
+            accWidth += item.width;
+            if (accWidth <= containerWidth) {
+                visibleItems.push(item);
+            } else if (!item.forceVisibility) {
+                hiddenItems.push(item);
+            }
+        }
```

Comment:
Here we do multiple iterations over one array. Also in subscribe callback another iteration over visible and hidden items. Can this be combined somehow? Maybe in `for` loop you can call `show` or `hide` methods?

Reply:
Another thing is that for `accWidth` you iterate over all items, even though you are interested only in items that were already filtered for the `visibleItems` variable.

Reply:
It's down to 2n now. I'd prefer to keep the subscribe code as-is

Reply:
I've thought about it more and

I need initially visibleItems to contain all the forceVisibleItems and to know how much is their width. If I start iterating and accumulate items, and overflow the allowed width and at the same time I have forceVisibility items left in the queue, I end up in the situation where I will have to iterate back and remove previously added items. I could not think of any better approach :/